src/services/image.service.ts

```diff
@@ -0,0 +1,34 @@
+import type { Playlist, PlaylistItem } from '#types/playlist';
+import { findPlaylistImageForWidth } from '#src/utils/collection';
+import type { Config } from '#types/Config';
+
+export type ImageProperty = 'shelfImage' | 'backgroundImage' | 'channelLogoImage';
+
+export const generateAlternateImageURL = (mediaid: string, imageLabel: string, width: number) =>
+  `https://img.jwplayer.com/v1/media/${mediaid}/images/${imageLabel}.jpg?width=${width}`;
+
+export const getAlternateImageLabel = (config: Config, propertyName: ImageProperty, item: PlaylistItem, playlist?: Playlist) => {
+  const alternateImageLabel = propertyName === 'shelfImage' ? playlist?.[propertyName] : item[propertyName];
+  const configImageLabel = config.custom?.[propertyName];
+
+  // image label override from playlist or playlist item
+  if (typeof alternateImageLabel === 'string') {
+    return alternateImageLabel;
+  }
+
+  // image label override from config
+  if (typeof configImageLabel === 'string') {
+    return configImageLabel;
+  }
```

Comment:
```suggestion
  return alternateImageLabel || configImageLabel;
```

Reply:
The `alternateImageLabel` and `configImageLabel` types are `unknown` so the explicit type check is used to satisfy typescript